poseidon2-circuit-air/src/columns.rs

```diff
@@ -0,0 +1,79 @@
+use core::borrow::{Borrow, BorrowMut};
+
+use p3_poseidon2_air::Poseidon2Cols;
+
+/// Columns for a Poseidon2 AIR which computes one permutation per row.
+///
+/// They extend the P3 columns with some circuit-specific columns.
+///
+/// `is_sponge` (transparent): if `1`, this row performs a sponge operation (absorb or squeeze);
+/// otherwise, it performs a compression.
+/// `reset` (transparent): indicates whether the state is being reset this row.
+/// `sponge_reset`: auxiliary column to keep constraint degrees below three.
+/// `absorb_flags` (transparent): for each rate element, indicates if it is being absorbed this row.
+/// At most one flag is set to 1 per row: if `absorb_flags[i]` is 1, then all elements up to the `i`-th
+/// are absorbed; the rest are propagated from the previous row.
+/// `input_indices` (transparent): for each input element, indicates the index in the witness table for the
+/// memory lookup. It's either received (for an absorb or a compression) or sent (for a squeeze).
+/// `output_indices` (transparent): for each output element, indicates the index in the witness table for the
+/// memory lookup. Only used by compressions to send the output.
+#[repr(C)]
+pub struct Poseidon2CircuitCols<
```

Comment:
Plonky3 uses a `vectorized` version of Poseidon2 air for optimal performances (prover side) with a default value of 8. While it's unclear with recursion what the best value is, it may be worth investigating the tradeoff and possibly upgrade the AIR here to account for this. Perhaps we can add a TODO for that?